lib/langchain/vectorsearch/base.rb

```diff
@@ -63,14 +63,13 @@ def generate_prompt(question:, context:)
       prompt_template.format(question: question)
     end
 
-    def add_data(path: nil, paths: nil)
-      raise ArgumentError, "Either path or paths must be provided" if path.nil? && paths.nil?
-      raise ArgumentError, "Either path or paths must be provided, not both" if !path.nil? && !paths.nil?
-
-      texts = Array(path || paths)
-        .flatten
```

Comment:
I think we still need `.flatten`, no? It covers the cases when someone passes in a 2d array:

```ruby
irb(main):002:0> Array([[]]).flatten
=> []
irb(main):003:0> Array([]).flatten
=> []
```

Reply:
I believe such cases rarely occur if we don't tell users to do so, but yeah we could keep `.flatten` to handle them. Will update the code tonight 😄 .

Reply:
@andreibondarev The code was updated. Please review it when you have time, thanks!